tensorflow-utils/nio-utils/src/main/java/org/tensorflow/nio/buffer/DataBuffers.java

```diff
@@ -0,0 +1,519 @@
+   */
+  public static BooleanDataBuffer ofBooleans(long size, BooleanDataAdapter adapter) {
+    return toBooleans(ofBytes(size * adapter.sizeInBytes()), adapter);
+  }
+
+  /**
+   * Adapt a physical buffer to a virtual buffer of booleans.
+   *
+   * <p>The provided adapter is used to create the boolean values to/from bytes, allowing custom
+   * representation of a boolean.
+   *
+   * @param buffer the buffer to adapt
+   * @param adapter an object converting buffer data to booleans
+   * @return a new buffer
+   */
+  public static BooleanDataBuffer toBooleans(ByteDataBuffer buffer, BooleanDataAdapter adapter) {
+    return BooleanVirtualDataBuffer.create(buffer, adapter);
+  }
+
+  /**
+   * Wraps an array of booleans into a data buffer.
+   *
+   * @param array array to wrap
+   * @param readOnly true if the buffer created must be read-only
+   * @return a new buffer
+   */
+  public static BooleanDataBuffer wrap(boolean[] array, boolean readOnly) {
+    if (unsafeAvailable) {
+      return BooleanRawDataBuffer.wrap(array, readOnly);
+    }
+    return BooleanArrayDataBuffer.wrap(array, readOnly);
+  }
+
+  /**
+   * Creates a buffer of objects of type {@code clazz` that can store up to `size} values
+   *
+   * @param type the type of object stored in this buffer
+   * @param size size of the buffer to allocate
+   * @return a new buffer
+   */
+  public static <T> DataBuffer<T> of(Class<T> type, long size) {
+    return ArrayDataBuffer.allocate(type, size);
+  }
+
+  /**
+   * Creates a virtual buffer that can store up to {@code size} values.
+   *
+   * <p>The provided adapter is used to create the values to/from bytes, allowing custom
+   * representation of this buffer type.
+   *
+   * @param size size of the buffer to allocate
+   * @return a new buffer
+   */
+  public static <T> DataBuffer<T> of(long size, DataAdapter<T> adapter) {
+    return to(ofBytes(size * adapter.sizeInBytes()), adapter);
+  }
+
+  /**
+   * Adapt a physical buffer to a virtual buffer.
+   *
+   * <p>The provided adapter is used to create the values to/from bytes, allowing custom
+   * representation of this buffer type.
+   *
+   * @param buffer the buffer to adapt
+   * @param adapter an object converting buffer data to booleans
+   * @return a new buffer
+   */
+  public static <T> DataBuffer<T> to(ByteDataBuffer buffer, DataAdapter<T> adapter) {
+    return VirtualDataBuffer.create(buffer, adapter);
+  }
+
+  /**
+   * Wraps an array of objects into a data buffer.
+   *
+   * @param array array to wrap
+   * @param readOnly true if the buffer created must be read-only
+   * @return a new buffer
+   */
+  public static <T> DataBuffer<T> wrap(T[] array, boolean readOnly) {
+    return ArrayDataBuffer.wrap(array, readOnly);
+  }
+
+  private static boolean unsafeAvailable;
+
+  static {
+    try {
+      Class.forName("sun.misc.Unsafe");
+      unsafeAvailable = true;
+    } catch (ClassNotFoundException e) {
+      unsafeAvailable = false;
```

Comment:
This is misleading. `sun.misc.Unsafe` is often available in other implementations of Java SE, but it doesn't always contain all the methods from OpenJDK. We need to make sure it has all the methods we want like this:
https://github.com/bytedeco/javacpp/blob/master/src/main/java/org/bytedeco/javacpp/indexer/UnsafeRaw.java

Reply:
Thanks, I'll revise this